source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRImageSetUICollectionView.mm

```diff
@@ -95,8 +95,25 @@ - (CGSize)intrinsicContentSize
 - (CGSize) collectionViewContentSize
 {
     size_t cellCounts = _imgSet->GetImages().size();
-    int dimension = (ceil(sqrt(cellCounts)));
     CGSize imageSize = ((UICollectionViewFlowLayout *)self.collectionViewLayout).itemSize;
-    return CGSizeMake(dimension * imageSize.width, dimension * imageSize.height);
+    float spacing = ((UICollectionViewFlowLayout *)self.collectionViewLayout).minimumInteritemSpacing;
+    float lineSpacing = ((UICollectionViewFlowLayout *)self.collectionViewLayout).minimumLineSpacing;
+
+    // sanity check
+    if(!imageSize.width || !self.frame.size.width || !cellCounts){
+        return CGSizeMake(0, 0);
+    }
+    int frameWidth = self.frame.size.width;
+    int imageWidthWithSpacing = imageSize.width + spacing;
+
+    // if there is spacing to the right edge, it's o.k.
+    int numbersOfItemsInRow = frameWidth / imageWidthWithSpacing;
```

Comment:
>int numbersOfItemsInRow = frameWidth / imageWidthWithSpacing; [](start = 4, length = 61)

I don't think this math is quite right, because of the truncation from float to int. Imagine you have a frame width of 100, image width of 10, and spacing of 0.5. You should be able to fit nine images into the frame, but this math will say you can fit 10.

Reply:
thanks, that's great find. really appreciate it.

---
In reply to: [197204662](https://github.com/Microsoft/AdaptiveCards/pull/1628#discussion_r197204662) [](ancestors = 197204662)